eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -114,6 +115,31 @@ object InteractiveTxBuilder {
     val minNextFeerate: FeeratePerKw = targetFeerate * 25 / 24
   }
 
+  // @formatter:off
+  sealed trait Purpose {
+    def previousLocalBalance: MilliSatoshi = 0.msat
+    def previousRemoteBalance: MilliSatoshi = 0.msat
+    def commitmentIndex: Long = 0
+    def perCommitmentPoint: PublicKey
+    def commitTxFeerate: FeeratePerKw
+    def common: Common
+  }
+  case class InitialCommitment(commitTxFeerate: FeeratePerKw, perCommitmentPoint: PublicKey, nextPerCommitmentPoint: PublicKey) extends Purpose {
+    override val common: Common = Common(
+      localChanges = LocalChanges(Nil, Nil, Nil), remoteChanges = RemoteChanges(Nil, Nil, Nil),
+      localNextHtlcId = 0L, remoteNextHtlcId = 0L,
+      localCommitIndex = 0L, remoteCommitIndex = 0L,
+      originChannels = Map.empty,
+      remoteNextCommitInfo = Right(nextPerCommitmentPoint),
+      remotePerCommitmentSecrets = ShaChain.init
+    )
+  }
+  case class RbfInitialCommitment(common: Common, commitment: Commitment) extends Purpose {
```

Comment:
Initially I went with:

```suggestion
  case class RbfInitialCommitment(common: Common, commitment: Commitment, previousTransactions: Seq[InteractiveTxBuilder.SignedSharedTransaction]) extends Purpose {
```

It is more pure (there is no ambiguity when rbfing) and works well, but would force to heavily refactor the tests, so I backtracked to get some feedback first.

Reply:
I see what you mean, the `aliceRbf` and `bobRbf` actors that are currently created directly in the fixture are a bit limiting here...it was a bit lazy to implement it this way, it could make sense to refactor this which would unblock more testing scenarios (e.g. more than 1 RBF attempt), and it would make sense to have the `previousTransactions` in this class. I think it would be a nice change, I'll let you decide depending on the amount of work necessary to rework the tests!

Reply:
Let me take a 2nd shot at it 🤓

Reply:
Refactored tests in 49c763809, and did the change in dbc8b6015. Pretty proud of that one!